src/api/Visualization/types.ts

```diff
@@ -175,11 +175,16 @@ export interface VisualizationThumbnail {
 }
 
 export interface VisualizationThumbnailNodeWithMetadata extends VisualizationThumbnailNode {
-  data: {categories: string[]};
+  data: {
+    categories: string[];
+    subCategories?: string[]; // In case of node group, the cateogories in this node group
```

Comment:
What if we have more than one nodeGrouping in the viz, do we group put all the categories from different rules in this list ?

Also i think we should rename `subCategories` to `nodeGroupingCategories` as the concept of `subCategories` does not  exist in the backend